File: Introduction/Week 2/problem/lcm/lcm.py
# Uses python3
import sys

def gcd_fast(a,b):
    val = [a,b]
    val.sort()
    a = val[1]
    b = val[0]

    if b == 0:
        return a
    else :
        return gcd_fast(a % b, b)

def lcm_fast(a, b):
    multiple = gcd_fast(a,b)
    
    return (a * b ) / multiple

if __name__ == '__main__':
    a, b = map(int, input().split())
    print(lcm_fast(a, b))

File: Introduction/Week 2/problem/lcm/lcm.cpp
#include <iostream>

long long gcd_fast(long long a, long long b) 
{
        long long temp = 0;

        if( a < b) {
                temp = b;
                b = a;
                a = temp;
        }

        if( b == 0) {
                return a;
        } else {
                return gcd_fast(a % b, b);
        }

}

long long lcm_fast(long long  a, long long b) 
{
        int multiple = 0;

        multiple = gcd_fast(a, b);

        return ((long long)(a * b)) / multiple ;
}

int main() {
  long long a, b;
  std::cin >> a >> b;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}
